app/Filament/Resources/MeetupResource.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Filament\Resources;
+
+use App\Filament\Resources\MeetupResource\Pages;
+use App\Filament\Resources\MeetupResource\RelationManagers;
+use App\Models\Group;
+use App\Models\Meetup;
+use Filament\Forms;
+use Filament\Forms\Components\Field;
+use Filament\Forms\Components\Select;
+use Filament\Forms\Form;
+use Filament\Resources\Resource;
+use Filament\Tables;
+use Filament\Tables\Table;
+use Illuminate\Database\Eloquent\Builder;
+
+class MeetupResource extends Resource
+{
+	protected static ?string $model = Meetup::class;
+	
+	protected static ?string $navigationIcon = 'heroicon-o-rectangle-stack';
+	
+	public static function getNavigationBadge(): ?string
+	{
+		return Meetup::query()
+			->when(
+				request()->attributes->get('group'),
+				fn(Builder $query, Group $group) => $query->where('group_id', $group->getKey())
+			)
+			->count();
+	}
+	
+	public static function form(Form $form): Form
+	{
+		return $form
+			->schema([
+				Forms\Components\Select::make('group_id')
+					->relationship(name: 'group', titleAttribute: 'name')
+					->columnSpanFull(), // FIXME
+				Forms\Components\TextInput::make('location')
+					->required()
+					->maxLength(255),
+				Forms\Components\TextInput::make('capacity')
+					->required()
+					->numeric(),
+				Forms\Components\DateTimePicker::make('starts_at')
+					->label('Start')
+					->required()
+					->rules(['required', 'date', 'before_or_equal:ends_at']),
+				Forms\Components\DateTimePicker::make('ends_at')
+					->label('End')
+					->required()
+					->rules(['required', 'date', 'after_or_equal:starts_at']),
+				Forms\Components\MarkdownEditor::make('description')
+					->required()
+					->columnSpanFull(),
```

Comment:
```suggestion
				Forms\Components\DateTimePicker::make('starts_at')
					->label('Start')
					->required()
					->beforeOrEqual('ends_at')
					->rules(['required', 'date']),
				Forms\Components\DateTimePicker::make('ends_at')
					->label('End')
					->required()
					->afterOrEqual('starts_at')
					->rules(['required', 'date']),
				Forms\Components\MarkdownEditor::make('description')
					->required()
					->columnSpanFull(),
```

This fixes the date window validation.

Reply:
Interesting. Do you know why?

Reply:
I'm not sure why using rules didn't  work too. It's something I have to dig in to when I have time.

Reply:
Looks like this is how it really behaves on filament. Check the overview of the validation docs.

https://filamentphp.com/docs/3.x/forms/validation

![CleanShot 2025-02-10 at 10 13 46@2x](https://github.com/user-attachments/assets/d495e7e7-33a1-485e-8a25-32b7ddc0aeab)